xbmc/addons/ScreenSaver.cpp

```diff
@@ -26,13 +23,9 @@ CScreenSaver::CScreenSaver(BinaryAddonBasePtr addonBase)
   m_profile = CSpecialProtocol::TranslatePath(Profile());
 
   m_struct = {{0}};
-#ifdef TARGET_WINDOWS
-  m_struct.props.device = DX::DeviceResources::Get()->GetD3DContext();
-#else
-  m_struct.props.device = nullptr;
-#endif
   m_struct.props.x = 0;
   m_struct.props.y = 0;
+  m_struct.props.device = CServiceBroker::GetWinSystem()->GetHWContext();
```

Comment:
I see no reason to keep this, an addon can use `GetHWContex()` now. the same for visualization.

Reply:
For me better to leave currently as backward compatibility for not changed addons.